src/RestWrite.js

```diff
@@ -429,7 +429,8 @@ RestWrite.prototype.handleAuthDataValidation = function (authData) {
       return Promise.resolve();
     }
     const validateAuthData = this.config.authDataManager.getValidatorForProvider(provider);
-    if (!validateAuthData) {
+    const authProvider = (this.config.auth || {})[provider] || {};
+    if (!validateAuthData || authProvider.enabled === false) {
```

Comment:
I think we could already add a deprecation warning and plan to require an explicit `enabled: true` to enable an adapter in the future. What do you think?